### PR TITLE
[feat] Feature flag system via config

### PR DESCRIPTION
**Pull request by xtinec**
Adding a feature flag system that is driven by superset_config.py. This change includes:
- Server side changes to specify a dedicated FEATURE_FLAG dictionary for listing feature flags. E.g.
```
FEATURE_FLAGS = { 'SCOPED_FILTER': true }
```
- Pass the new feature flags to client via bootstrap-data
- Client side changes to inject feature flags into the redux state tree for dashboard, explore view and SqlLab
- Client side refactor/clean up so the feature flags can be properly tested. Also avoid modifying incoming bootstrap data when creating initial state for the redux state tree
- Re-enable tests that were previously disabled for ExploreViewContainer

@mistercrunch @kristw @williaster

### CODE REVIEW

File: superset/assets/src/featureFlags.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// A higher-order function that takes the redux state tree and returns a
+// `isFeatureEnabled` function which takes a feature and returns whether it is enabled.
+// Note that we assume the featureFlags subtree is at the root of the redux state tree.
+export function isFeatureEnabledCreator(state) {
+  return feature => !!state.featureFlags[feature];
+}
+
+// Feature flags are not altered throughout the life time of the app
+export default function featureFlagsReducer(state = {}) {
```
Comment (williaster) [inline superset/assets/src/featureFlags.js]: do you think this is necessary if it's never modified?

Comment (xtinec) [reply superset/assets/src/featureFlags.js]: I think it feels odd to have a subtree of the redux state tree dangling without a reducer. This also leaves room for enriching the feature flag system in the future if we'd like (e.g. allowing toggling of a feature flag without restarting the server). Thoughts?

File: superset/assets/src/featureFlags.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// A higher-order function that takes the redux state tree and returns a
+// `isFeatureEnabled` function which takes a feature and returns whether it is enabled.
+// Note that we assume the featureFlags subtree is at the root of the redux state tree.
+export function isFeatureEnabledCreator(state) {
+  return feature => !!state.featureFlags[feature];
```
Comment (mistercrunch) [inline superset/assets/src/featureFlags.js]: This makes it clear to me that feature flag can only be boolean, which seems reasonable given the name. It prevents having more complex data structures like `vizTypeBlacklist = ['heatmap', 'sunburst']` .  I'm not necessarily  questioning this, just pointing it out.

File: superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
```
Comment (mistercrunch) [inline superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.jsx]: I think you didn't use `git mv` and git/github thinks you created a new file, let's keep the git history in place

File: superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.js [hunk 4]
```diff
@@ -1,38 +0,0 @@
-// this test must be commented out because ChartContainer is now importing files
```
Comment (mistercrunch) [inline superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.js]: I'm confused as to why this was commented out to start with (?)

Comment (xtinec) [reply superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.js]: Looks like it predates the history on github
https://github.com/apache/incubator-superset/commits/master/superset/assets/spec/javascripts/explore/components/ExploreViewContainer_spec.js
Guess we'll never know... 🤔

File: superset/assets/src/featureFlags.js (hunk 1 above)
Comment (kristw) [reply superset/assets/src/featureFlags.js]: Feature flag override would be interesting. Some apps has Chrome plugin that you can turn on/off experimental features. With that said I am OK with having empty reducer here for now.

**Review by kristw — APPROVED** [review]

**Comment by xtinec** [comment]
@mistercrunch @williaster does this look good to merge? 🙏

**Review by williaster — APPROVED** [review]
lgtm!

**Pull request merged**